routes/article.js

```diff
@@ -21,53 +21,37 @@ function getAmount(vote) {
 
 router.post('/vote', (req, res) => {
   const amount = getAmount(req.body.vote)
-  if (amount === 0) {
+  if (!amount) {
     res.status(500).send({
       err: 'Invalid value for vote',
       givens: req.body,
     })
     return
   }
 
-  let articleChanged = null
   Article.findById(req.body.id)
-    .then((article) => {
-      article.votes += amount
-      articleChanged = article
-      return article.save()
-    })
-    .then((data) => {
-      data.keywords.forEach((word) => {
-        Keyword.findOne({ word })
-          .then((keyword) => {
-            const votes = keyword.votes
-            const len = votes.length
-            const newVote = {
-              sum: votes[len - 1].sum + amount,
-              time: Date.now(),
-            }
-            votes.push(newVote)
-            keyword.save()
-          })
-          .catch((err) => {
-            const keyword = new Keyword({
-              word,
-              votes: [{
-                sum: amount,
-                time: Date.now(),
-              }],
-            })
-            keyword.save()
+        .then((article) => {
+          if (!article) throw new Error('Article not found')
+          article.votes += amount
+          return article.save()
```

Comment:
A better approach would be to use `findByIdAndUpdate`, so there would only be one request sent to the database server inside of querying the database, getting and modifying the response and then requesting to save. Also then no modifications to params 😎 
```js
Article.findByIdAndUpdate(
  req.body.id,
  { $inc : { votes: amount } },
  {
    new: true,
    runValidators: true
   })
   .then((article) => ...
```

Heres the docs : http://mongoosejs.com/docs/api.html#model_Model.findByIdAndUpdate